src/hazard/onboard/tudelft_flood.py

```diff
@@ -127,68 +149,94 @@ def run_single(self, item: BatchItem, source: Any, target: ReadWriteDataArray, c
+                z_sop = target.create_empty(
+                    self._sop_resource.path.format(scenario=item.scenario, year=item.central_year),
+                    shape[1],
+                    shape[0],
+                    da_sop.rio.transform(),
+                    str(da_sop.crs),
+                    index_name="standard of protection (years)",
+                    indexes=["min", "max"],
+                )
+                values_max_sop = np.array(da_sop.data, dtype="float32")
+                sop_no_data = da_sop.attrs["nodatavals"]
+                values_max_sop[values_max_sop == sop_no_data] = float("nan")
+                values_min_sop = self._get_mins(values_max_sop)
+                z_sop[0, 0 : len(da_sop.y), 0 : len(da_sop.x)] = values_min_sop[:, :]
+                z_sop[1, 0 : len(da_sop.y), 0 : len(da_sop.x)] = values_max_sop[:, :]
+                del values_min_sop, values_max_sop
+
         for i, return_period in enumerate(self.return_periods):
             full_path_depth = str(full_path_depth_format).format(return_period=self.return_period_str[return_period])
             with self.fs.open(full_path_depth, "rb") as fd:
-                dad = xr.open_rasterio(fd).isel(band=0)
-                with self.fs.open(full_path_extent, "rb") as fe:
-                    dae = xr.open_rasterio(fe).isel(band=0)
-                    # bounds = da.rio.bounds()
-                    if return_period == self.return_periods[0]:
-                        z = target.create_empty(
-                            self._resource.path.format(scenario=item.scenario, year=item.central_year),
-                            shape[1],
-                            shape[0],
-                            dad.rio.transform(),
-                            str(dad.crs),
-                            indexes=self.return_periods,
-                        )
-                # dad_nodata = 65535
-                if (
-                    dad.shape[1] == 38371
-                ):  # corrections for various possible errors whereby coordinates are missing for certain files:
-                    dae = dae[:, 0:38371]
-                if dae.shape[1] == 38375:
-                    dae = dae[:, 0:38374]
-                if dad.shape[1] == 38375:
-                    dad = dad[:, 0:38374]
-                if dae.shape[0] == 39385:
-                    dad = dad[35:, :]
-
+                da_depth = xr.open_rasterio(fd).isel(band=0)
+                if return_period == self.return_periods[0]:
+                    z_depth = target.create_empty(
+                        self._depth_resource.path.format(scenario=item.scenario, year=item.central_year),
+                        shape[1],
+                        shape[0],
+                        da_depth.rio.transform(),
+                        str(da_depth.crs),
+                        indexes=self.return_periods,
+                    )
+                if da_depth.shape[1] == 38375:
```

Comment:
This is a question related with PR 108 (and originally from [PR 27](https://github.com/os-climate/hazard/pull/27#issuecomment-1933536976) 

If you had a minute to explain the 38375/38374 change we would appreciate it.